.circleci/config.yml

```diff
@@ -435,9 +351,163 @@ commands:
 
                   ./scripts/submit-codecov-data.sh
 
-      # NOTE: We store the logs to ease with troubleshooting / debugging
-      - store_artifacts:
-          path: ~/artifacts
+            # NOTE: We store the logs to ease with troubleshooting / debugging
+            - store_artifacts:
+                path: ~/artifacts
+
+  init-cached-docker-image:
+    parameters:
+      cache-key-prefix:
+        type: string
+      cache-path:
+        type: string
+      image-builder-path:
+        type: string
+      python_executable_path:
+        type: string
+        default: "/usr/bin/env python"
+    steps:
+      - run:
+          name: "Get checksum file for builder '<< parameters.image-builder-path >>'."
+          command: |
+            python -m << parameters.image-builder-path >> --checksum > ~/image_checksum
+
+      - restore_cache:
+          name: "Restore docker image file for the builder '<< parameters.image-builder-path >>'."
+          key: << parameters.cache-key-prefix >>-{{ .Branch }}-{{ checksum "~/image_checksum" }}-<< pipeline.parameters.cache_version >>
+
+      - run:
+          name: "Build image by '<< parameters.image-builder-path >>' builder."
+          command: |
+             << parameters.python_executable_path >> -m << parameters.image-builder-path >> --build-with-cache << parameters.cache-path >>
+
+      - save_cache:
+          name: "Save docker image of the builder '<< parameters.image-builder-path >>' into cache."
+          key: << parameters.cache-key-prefix >>-{{ .Branch }}-{{ checksum "~/image_checksum" }}-<< pipeline.parameters.cache_version >>
+          paths:
+            - << parameters.cache-path >>
+
+      - run:
+          name: "Remove image checksum file."
+          command: |
+            rm -f ~/image_checksum
+
+  install-cached-tox:
+    parameters:
```

Comment:
Please add description attribute here which describes what that command does / what kind of purposes it serves.